packages/react-kit/src/components/DateInput/DateInput.tsx

```diff
@@ -238,7 +239,16 @@ class RawDateInput extends React.Component<TDateInputFullProps, TDateInputState>
 			}
 			case ActiveSection.Month: {
 				//month starts from 1 and cannot be zero
-				const newMonth = month.map(value => (value - 1) % 13 || 12).orElse(() => some(12));
+				const newMonth = month
+					.map(value => {
+						if (value - 1 === 0) {
```

Comment:
This bunch of `if`s may be replaced with `.map(value => (value - 1 + 12) % 12)`

Reply:
we could also add some tests